### ISSUE TITLE
significant cpu usage

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/webdev#585]
I'm seeing significant cpu usage when debugging a flutter web app. The repro steps are:

- launch in IntelliJ
- select a web / chrome device
- set a breakpoint
- run your Flutter app in debug mode
- the apps runs; when the breakpoint is hit, the execution location doesn't show for perhaps 30 seconds, and the cpu core running the dart process is pegged at 100%. I haven't dug into the issue, but perhaps we're doing some expensive work per script (and a flutter app generally has 600-900 scripts)

**Comment by jakemac53** [comment]
@alan-knight @grouma can one of you take a look at this?

I wouldn't be surprised if all the source_map work we have done for breakpoints is causing this, I think it is pretty eager about collecting source information for the entire app?

It likely needs to be changed to be as lazy as possible.

**Comment by grouma** [comment]
Are you debugging through `webdev serve --debug` or debugging directly through IntelliJ? It would be good to limit the number of variables to consider.

**Comment by devoncarew** [comment]
I'm debugging a flutter web app via the flutter daemon equivalent of `flutter run -d chrome`, using IntelliJ.

Happy to provide more details about setup and repro steps, as well as the protocol traffic, if its useful.

**Comment by grouma** [comment]
I was able to reproduce this issue with the Flutter Gallery app. However, the Flutter CLI is using an old version of `package:dwds`. It is currently stuck on `0.5.0`. We made a number of improvements in version `0.5.1`, e.g. https://github.com/dart-lang/webdev/pull/544.

When running the Flutter Gallery with `webdev` at head I don't see any performance issues. @jonahwilliams said he has worked through the issues preventing Flutter from depending on the latest `package:dwds` and those changes should land soon. I'm going to close this issue for now.
